### PR TITLE
Fix AspNetCore preview7 break changes

### PR DESCRIPTION
**Pull request by JialinXin**
#584

### CODE REVIEW

File: src/Microsoft.Azure.SignalR/HubHost/ServiceHubConnectionContext.cs [hunk 1]
```diff
@@ -30,10 +30,19 @@ internal class ServiceHubConnectionContext : HubConnectionContext
         public const string ItemsUnavailableError = nameof(Items) + UnavailableErrorTemplate;
         public const string ProtocolUnavailableError = nameof(Protocol) + UnavailableErrorTemplate;
 
+#if NETCOREAPP3_0
+        public ServiceHubConnectionContext(HttpContext context)
```
Comment (davidfowl) [inline src/Microsoft.Azure.SignalR/HubHost/ServiceHubConnectionContext.cs]: cc @BrennanConroy @mikaelm12 did we break this?

File: test/Microsoft.Azure.SignalR.Tests/Infrastructure/DefaultHubProtocolResolver.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.SignalR.Protocol;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Microsoft.AspNetCore.SignalR.Internal
+{
+    // copied from https://github.com/aspnet/AspNetCore/blob/release/3.0-preview7/src/SignalR/server/Core/src/Internal/DefaultHubProtocolResolver.cs
+    internal class DefaultHubProtocolResolver : IHubProtocolResolver
```
Comment (davidfowl) [inline test/Microsoft.Azure.SignalR.Tests/Infrastructure/DefaultHubProtocolResolver.cs]: Where is this used?

Comment (davidfowl) [reply test/Microsoft.Azure.SignalR.Tests/Infrastructure/DefaultHubProtocolResolver.cs]: cc @anurse

Comment (davidfowl) [reply test/Microsoft.Azure.SignalR.Tests/Infrastructure/DefaultHubProtocolResolver.cs]: Ah tests 😄

File: src/Microsoft.Azure.SignalR/HubHost/ServiceHubConnectionContext.cs (hunk 1 above)
Comment (BrennanConroy) [reply src/Microsoft.Azure.SignalR/HubHost/ServiceHubConnectionContext.cs]: Yes
https://github.com/aspnet/AspNetCore/issues/11431

File: test/Microsoft.Azure.SignalR.Tests/Infrastructure/DefaultHubProtocolResolver.cs (hunk 2 above)
Comment (JialinXin) [reply test/Microsoft.Azure.SignalR.Tests/Infrastructure/DefaultHubProtocolResolver.cs]: In UT here, https://github.com/Azure/azure-signalr/blob/dev/test/Microsoft.Azure.SignalR.Tests/ServiceLifetimeManagerFacts.cs#L30

File: src/Microsoft.Azure.SignalR/HubHost/ServiceHubConnectionContext.cs (hunk 1 above)
Comment (davidfowl) [reply src/Microsoft.Azure.SignalR/HubHost/ServiceHubConnectionContext.cs]: Ah yes.

**Pull request merged**